lib/generator.js

```diff
@@ -86,13 +87,31 @@ class Generator {
    * @param {Boolean} [options.install=false] Install the template and its dependencies, even when the template has already been installed.
    * @param {Boolean} [options.debug=false] Enable more specific errors in the console. At the moment it only shows specific errors about filters. Keep in mind that as a result errors about template are less descriptive.
    * @param {Object<String, String>} [options.mapBaseUrlToFolder] Optional parameter to map schema references from a base url to a local base folder e.g. url=https://schema.example.com/crm/  folder=./test/docs/ .
+   * @param {Object} [options.registry]           Optional parameter with private registry configuration
+   * @param {String} [options.registry.url]       Parameter to pass npm registry url
+   * @param {String} [options.registry.username]  Optional parameter to pass npm registry username
+   * @param {String} [options.registry.password]  Optional parameter to pass npm registry base64 encoded password
+   * @param {String} [options.registry.token]     Optional parameter to pass npm registry auth token
    */
-  constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false, mapBaseUrlToFolder = {} } = {}) {
-    const invalidOptions = getInvalidOptions(GENERATOR_OPTIONS, arguments[arguments.length - 1] || []);
+  
+
+  constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false, mapBaseUrlToFolder = {}, registry = {url: '', username: '', password: '', token: '' }} = {}) {
```

Comment:
```suggestion
  constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false, mapBaseUrlToFolder = {}, registry = {url: '', username: '', password: '', token: '' }} = {}) {
```

can't we just do `registry = {}` ?

Reply:
Sure, done @derberg